packages/tenant-process/test/revokeVerifiedAttribute.test.ts

```diff
@@ -0,0 +1,323 @@
+      ],
+    };
+
+    await addOneTenant(tenantWithoutSameAttributeId);
+    await addOneTenant(revokerTenant);
+    await writeInReadmodel(toReadModelEService(eService), eservices);
+    await writeInReadmodel(toReadModelAgreement(agreementEservice), agreements);
+    expect(
+      tenantService.revokeVerifiedAttribute(
+        {
+          tenantId: tenantWithoutSameAttributeId.id,
+          attributeId: verifiedAttribute.id,
+        },
+        {
+          authData,
+          correlationId: generateId(),
+          serviceName: "",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(attributeNotFound(verifiedAttribute.id));
+  });
+  it("Should throw attributeRevocationNotAllowed if the organization is not allowed to revoke the attribute", async () => {
+    const tenantWithVerifiedAttribute: Tenant = {
+      ...targetTenant,
+      attributes: [
+        {
+          ...verifiedAttribute,
+          verifiedBy: [
+            {
+              ...getMockVerifiedBy(),
+              id: generateId(),
+            },
+          ],
+          revokedBy: [{ ...getMockRevokedBy() }],
+        },
+      ],
+    };
+
+    await addOneTenant(tenantWithVerifiedAttribute);
+    await addOneTenant(revokerTenant);
+    await writeInReadmodel(toReadModelEService(eService), eservices);
+    await writeInReadmodel(toReadModelAgreement(agreementEservice), agreements);
+
+    expect(
+      tenantService.revokeVerifiedAttribute(
+        {
+          tenantId: tenantWithVerifiedAttribute.id,
+          attributeId: verifiedAttribute.id,
+        },
+        {
+          authData,
+          correlationId: generateId(),
+          serviceName: "",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(
+      attributeRevocationNotAllowed(targetTenant.id, verifiedAttribute.id)
+    );
+  });
+  it("Should throw verifiedAttributeSelfRevocationNotAllowed if the organizations are not allowed to revoke own attributes", async () => {
+    await addOneTenant(revokerTenant);
+    await writeInReadmodel(toReadModelEService(eService), eservices);
+    await writeInReadmodel(toReadModelAgreement(agreementEservice), agreements);
+
+    expect(
+      tenantService.revokeVerifiedAttribute(
+        {
+          tenantId: revokerTenant.id,
+          attributeId: verifiedAttribute.id,
+        },
+        {
+          authData,
+          correlationId: generateId(),
+          serviceName: "",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(verifiedAttributeSelfRevocationNotAllowed());
+  });
+  it("Should throw attributeAlreadyRevoked if the attribute is already revoked", async () => {
+    const tenantWithVerifiedAttribute: Tenant = {
+      ...targetTenant,
+      attributes: [
+        {
+          ...verifiedAttribute,
+          verifiedBy: [
+            {
+              ...getMockVerifiedBy(),
+              id: revokerTenant.id,
+            },
+          ],
```

Comment:
According to the recent discussion, when a `tenant A` revokes an attribute to `tenant B`, then `tenant A` is no longer inside the `verifiedBy` array of `tenant B`. So in this case, it should be here.
By the way, when you change this you will probably notice an error, because this test would raise `attributeRevocationNotAllowed` because this check happens first. To fix this, swap the position the two checks in the business logic.